diff --git a/src/lj_prng.c b/src/lj_prng.c
index 6e5a9921..0da7ff02 100644
--- a/src/lj_prng.c
+++ b/src/lj_prng.c
@@ -87,6 +87,10 @@ extern int sys_get_random_number(void *buf, uint64_t len);
 
 extern int sceRandomGetRandomNumber(void *buf, size_t len);
 
+#elif defined(__SWITCH__)
+
+#include <switch.h>
+
 #elif LJ_TARGET_NX
 
 #include <unistd.h>
@@ -180,6 +184,11 @@ int LJ_FASTCALL lj_prng_seed_secure(PRNGState *rs)
   if (sceRandomGetRandomNumber(rs->u, sizeof(rs->u)) == 0)
     goto ok;
 
+#elif defined(__SWITCH__)
+
+  randomGet(rs->u, sizeof(rs->u));
+  goto ok;
+
 #elif LJ_TARGET_NX
 
   if (getentropy(rs->u, sizeof(rs->u)) == 0)
